spec/models/asset_spec.rb

```diff
@@ -198,6 +199,79 @@
 
       asset.scanned_clean!
     end
+
+    context 'when asset is already clean' do
+      let(:state) { 'clean' }
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_clean! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+
+    context 'when asset is already infected' do
+      let(:state) { 'infected' }
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_clean! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+
+    context 'when asset is already uploaded' do
+      let(:state) { 'uploaded' }
+
+      # before do
+      #   asset.file = load_fixture_file('asset.png')
+      # end
```

Comment:
It looks as though you can remove this commented out `before` block.

Reply:
@chrisroos Well spotted - thanks!